CPS-????/README.md

```diff
@@ -0,0 +1,173 @@
+---
+CPS: ?
+Title: Intents for Cardano
+Status: Open
+Category: ?
+Authors:
+    - Michael Peyton Jones <michael.peyton-jones@iohk.io>
+Proposed Solutions: []
+Discussions:
+Created: 2024-03-11
+License: CC-BY-4.0
+---
+
+## Abstract
+
+We describe a family of issues around “intents” on Cardano. We do some basic analysis on the problem space, and suggest some high-level areas for investigation.
+
+## Problem
+
+In many cases, a user’s desired interaction with a system is not fully specified. 
+Classic examples include things like “market orders” where a user wants to buy an asset at the “market price”, and does not specify exactly what that is, leaving it up to another system to resolve. 
+A common name for these kinds of “incomplete” or “partial” actions is “intents”. 
+We refer to the process of turning an intent into something complete as “fulfilment”.
+
+Intents are [quite prevalent](https://members.delphidigital.io/reports/wtf-is-anoma-part-2-wtf-are-intent-based-apps) in blockchain systems (see “Use cases” for more discussion of specific examples), but they are also problematic since blockchain systems usually require a fairly high degree of specificity in their settlement layer. 
+For example, if you want to perform a trade, you usually need to specify exactly who with and exactly how much for.
+
+Cardano is unusually stringent in this regard. 
+On Ethereum, for example, smart contracts can actually change what your transaction does, in a way that is only determined when the transaction is validated. 
+In contrast, on Cardano smart contracts only validate the behaviour that the user has specified in their transaction. 
+This gives us very nice determinism properties... but when resolving intents we actually want non-determinism!
+
+We can see the “concurrency problem” through this lens. 
+Cardano insists that state updates (via datums in UTXOs) are completely explicit and deterministically sequenced, in contrast to systems like Ethereum, where state access is implicit and non-deterministically sequenced. 
+This is quite inconvenient, since what users often want is to interact with “the current state” of an application, and they are happy for this to be determined non-deterministically.[^concurrency]
+
+[^concurrency]: See https://github.com/input-output-hk/Developer-Experience-working-group/issues/47 for discussion of a very intent-like approach to solving the concurrency problem.
+
+To support intent-based interactions, we need two things:
```

Comment:
we (obviously) also need a language for expressing intents and their fulfillments